planet/client.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+from planet.clients.data import DataAPI
+from planet.clients.orders import OrdersAPI
+from planet.clients.subscriptions import SubscriptionsAPI
+from planet.http import Session
+
+SYNC_CLIENT_AGENT = "python-sdk-sync"
+
+
+class Planet:
+    """Planet API client. This client contains non-async methods.
+
+    Authentication is required: defaults to detecting API key from environment (PL_API_KEY).
+
+    Members:
+    `data`: for interacting with the Planet Data API.
+    `orders`: Orders API.
+    `subscriptions`: Subscriptions API.
+
+    Quick start example:
+    ```python
+    # requires PL_API_KEY
+
+    pl = Planet()
+    for item in pl.data.search(['PSScene'], limit=5):
+        print(item)
+
+    for sub in pl.subscriptions.list_subscriptions():
+        print(item)
+    ```
+
+    Parameters:
+        session: Optional Session. If not provided, a new session is created.
+          The session can be used to control the authentication method. Example:
+
+          ```python
+          from planet import Auth, Session, Planet
+
+          auth = Auth.from_key('examplekey')
+          session = Session(auth=auth)
+          pl = Planet(session=session)
+          ````
+    """
+
+    def __init__(self, session: Optional[Session] = None) -> None:
+        self._session = session or Session()
+        self._session._client.headers.update(
+            {"X-Planet-App": SYNC_CLIENT_AGENT})
+
+        self.data = DataAPI(self._session)
+        self.orders = OrdersAPI(self._session)
+        self.subscriptions = SubscriptionsAPI(self._session)
```

Comment:
Do all of these clients need to be instantiated at the same time?  I wonder if there is a way to 'opt-in' to the individual clients, e.g. a Subscriptions API user who only wants to use the Subscriptions client.  As coded, this user will absolutely be able to use the Subscriptions client, but they will also instantiate Data API and Orders API clients.

Reply:
I don't think there's much overhead for an end user to have these classes instantiated behind the scenes when the Planet client is created. They're really just providing methods and not holding any state. I think my question would be if there's any benefit to being able to opt-in/out of Data/Orders/Subscriptions on the client- there should be no harm in a user creating a Planet class and ignoring e.g. subscriptions until they need it (IMO).